ngraph/core/include/ngraph/partial_shape.hpp

```diff
@@ -223,6 +230,18 @@ namespace ngraph
                                          const PartialShape& src,
                                          const op::AutoBroadcastSpec& autob);
 
+        iterator begin() noexcept { return m_dimensions.begin(); }
```

Comment:
Just one comment: Please add doxygen documentation for new methods.

Reply:
Done

Reply:
@ilyachur, should we change `m_shape_type` as is done in `operator[]`? 
IMHO `begin`, `end` and `operator[]` should return some wrapper which on each write should update `m_shape_type`.